File: ruby/hamming/hamming.rb
class Hamming
  def self.compute(str_one, str_two)
    count = str_one.length
    if (str_one.length == str_two.length)
    for i in 0..str_one.length-1
      count -= 1 if (str_one[i] == str_two[i])
    end
    count
  else
    raise(ArgumentError)
  end
  end
end

File: javascript/hamming/hamming.js
var Hamming = function() {};

Hamming.prototype.compute = function(dna, dnaTwo ) {
  if (dna.length != dnaTwo.length){
    throw new Error('DNA strands must be of equal length.');
    return;
  }
  var count = 0;
  for (var i = 0; i < dna.length; i++){
    if (dna.charAt(i) != dnaTwo.charAt(i)){
      count++;
    }
  }
  return count;
};

module.exports = Hamming;
